magefile.go

```diff
@@ -1077,101 +1077,126 @@ func collectPackageDependencies(platforms []string, packageVersion string, requi
+		for _, rp := range requiredPackages {
+			files, err := filepath.Glob(filepath.Join(sourcePath, "*"+rp+"*"))
+			if err != nil {
+				panic(err)
+			}
 
-					if err := cmd.Run(); err != nil {
-						panic(err)
-					}
+			targetPath := filepath.Join(archivePath, rp)
+			os.MkdirAll(targetPath, 0755)
+			for _, f := range files {
+				// safety check; if the user has an older version of the beats repo,
+				// for example right after a release where you've `git pulled` from on repo and not the other,
+				// they might end up with a mishmash of packages from different versions.
+				// check to see if we have mismatched versions.
+				if !strings.Contains(f, packageVersion) {
+					// if this panic hits weird edge cases where we don't want actual failures, revert to a printf statement.
+					panic(fmt.Sprintf("the file %s doesn't match agent version %s, beats repo might be out of date", f, packageVersion))
 				}
 
-				// copy to new drop
-				sourcePath := filepath.Join(pwd, "build", "distributions")
-				for _, rp := range requiredPackages {
-					files, err := filepath.Glob(filepath.Join(sourcePath, "*"+rp+"*"))
-					if err != nil {
-						panic(err)
-					}
-
-					targetPath := filepath.Join(archivePath, rp)
-					os.MkdirAll(targetPath, 0755)
-					for _, f := range files {
-						// safety check; if the user has an older version of the beats repo,
-						// for example right after a release where you've `git pulled` from on repo and not the other,
-						// they might end up with a mishmash of packages from different versions.
-						// check to see if we have mismatched versions.
-						if !strings.Contains(f, packageVersion) {
-							// if this panic hits weird edge cases where we don't want actual failures, revert to a printf statement.
-							panic(fmt.Sprintf("the file %s doesn't match agent version %s, beats repo might be out of date", f, packageVersion))
-						}
-
-						targetFile := filepath.Join(targetPath, filepath.Base(f))
-						packagesCopied += 1
-						if err := sh.Copy(targetFile, f); err != nil {
-							panic(err)
-						}
-					}
-				}
-				// a very basic footcannon protector; if packages are missing and we need to rebuild them, check to see if those files were copied
-				// if we needed to repackage beats but still somehow copied nothing, could indicate an issue. Usually due to beats and agent being at different versions.
-				if packagesCopied == 0 {
-					fmt.Println(">>> WARNING: no packages were copied, but we repackaged beats anyway. Check binary to see if intended beats are there.")
+				targetFile := filepath.Join(targetPath, filepath.Base(f))
+				packagesCopied += 1
+				if err := sh.Copy(targetFile, f); err != nil {
+					panic(err)
 				}
 			}
 		}
-	} else {
-		archivePath = movePackagesToArchive(dropPath, requiredPackages)
+		// a very basic footcannon protector; if packages are missing and we need to rebuild them, check to see if those files were copied
```

Comment:
Can all of this be a separate PR with a description for what it fixes? Doesn't seem related to the endpoint change.